azure.go

```diff
@@ -270,29 +272,27 @@ func (ac *AzureClient) listByTag(tagName string, tagValue string, types []string
 		return nil, fmt.Errorf("Error unmarshalling response body: %v", err)
 	}
 	if len(types) > 0 {
-		typesMap := make(map[string]struct{})
-		for _, resourceType := range types {
-			typesMap[resourceType] = struct{}{}
-		}
-		var filteredResources []struct {
-			Id        string `json:"id"`
-			Name      string `json:"name"`
-			Type      string `json:"type"`
-			ManagedBy string `json:"managedBy"`
-			Location  string `json:"location"`
-		}
-		for _, resource := range data.Value {
-			if _, typeExist := typesMap[resource.Type]; typeExist {
-				filteredResources = append(filteredResources, resource)
-			}
-		}
-		data.Value = filteredResources
+		filterTypesInResourceList(types, data)
 	}
 	resources := extractResourceNames(data, subscription)
 
 	return resources, nil
 }
 
+func filterTypesInResourceList(types []string, data AzureResourceListResponse) {
```

Comment:
I would make two changes here but would like your input on them too:

1. change the function to a method as it's specific to the `AzureResourceListResponse` object.
2. return the filtered list instead of changing the original

Reply:
I agree, at least for #2!

Reply:
1. do you mean : func (r *AzureResourceListResponse) filterTypes ?

Reply:
@vincentcreusotfx yes exactly

Reply:
ok modified ... but not sure if I should not modify the receiver value in the method directly

Reply:
Can you be more precise @vincentcreusotfx ? De you really mean the `receiver value`, or you mean the `receiver value`? If you mean `received value`, why not: other filtering function in the project return a filtered list instead of changing the original no?

Reply:
Talking about response.Value to update directly in the method instead of after receiving it back

Reply:
.....other filtering function in the project return a filtered list instead of changing the original no?

Reply:
Yes but the method is on the AzureResponseList "object" so we call data.filter() then we modify the data value ...

Reply:
Yes but the function is called on the data object and after that we modify the data value ...

Reply:
But anyway its a detail